CONTRIBUTING.rst

```diff
@@ -162,3 +162,134 @@ their individual contributions.
 * `Will Hall <https://www.github.com/wrhall>`_ (`wrsh07@gmail.com <mailto:wrsh07@gmail.com>`_)
 * `Will Thompson <https://www.github.com/wjt>`_ (`will@willthompson.co.uk <mailto:will@willthompson.co.uk>`_)
 * `Zac Hatfield-Dodds <https://www.github.com/Zac-HD>`_ (`zac.hatfield.dodds@gmail.com <mailto:zac.hatfield.dodds@gmail.com>`_)
+
+
+-----------------
+Review Guidelines
+-----------------
+
+All changes to Hypothesis go through a code review process, and must be
+signed off by at least one person with write access to the repo other
+than the author of the change. This document
+outlines some guidelines on what to look for and what we're interested
+in. It applies uniformly to maintainers and external contributors.
+
+At a high level, the two things we're looking for in review are answers
+to the following questions:
+
+1. Is this change going to make users' lives worse?
+2. Is this change going to make the maintainers' lives worse?
+
+Code review is a collaborative process between the author and the
+reviewer to try to ensure that the answer to both of those questions
+is no.
+
+Ideally of course the change should also make one or both of the users'
+and our lives *better*, but it's OK for changes to be mostly neutral.
+The author should be presumed to have a good reason for submitting the
+change in the first place, so neutral is good enough!
+
+The rest of this section outlines specific questions reviewers should
+ask in aid of this, but the above is the most important part and where
+the guidelines below clash with the objectives above, the above wins
+out.
+
+When submitting changes it can be helpful to go through a pre-review
+pass of this yourself just to see what people will look at, but it's
+not required.
+
+~~~~~~~~~~~~~~~~~
+General Questions
+~~~~~~~~~~~~~~~~~
+
+The following are useful questions to ask for almost every change:
+
+1. Is this change too large? Could it profitably be broken up into
+   several smaller changes that could be judged on their individual
+   merits?
+2. Is the motivation for this change clear?
+3. If there is code in this change, does it make sense to at least
+   one person who isn't the author? 
+4. Is this change well covered by the review guidelines and is there
+   anything that could usefully be added to the guidelines to improve
+   that?
+
+~~~~~~~~~~~
+API Changes
+~~~~~~~~~~~
+
+Public API changes require the most careful scrutiny of all reviews,
+because they are the ones we are stuck with for the longest: Hypothesis
+follows semver and additionally tries not to break API compatibility
+wherever possible.
+
+When a public API changes we should ask the following questions:
+
+1. Is this change backwards compatible?
+2. Are ther eany backwards incompatible changes we'd want to make to this
+   API later and if so can we do them now instead? 
+3. How does this API handle invalid input and are the error messages clear
+   when that happens?
```

Comment:
You and I both have a policy that error messages should include the bad input, if/when it occurs. Worth stating here as an explicit principle?